apps/system/test/unit/sync_state_machine_test.js

```diff
@@ -0,0 +1,258 @@
+/* -*- Mode: Java; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+/* global SyncStateMachine */
+/* global SyncStateMachineEvents */
+
+'use strict';
+
+requireApp('system/js/sync_state_machine.js');
+
+suite('system/SyncStateMachine >', () => {
+  suite('Initial state', () => {
+    test('Integrity', () => {
+      // SyncStateMachine is supposed to expose a function per event.
+      for (var name in SyncStateMachineEvents) {
+        if (!SyncStateMachineEvents.hasOwnProperty(name)) {
+          continue;
+        }
+        assert.ok(SyncStateMachine[name]);
+        assert.ok(SyncStateMachine[name] instanceof Function);
+      }
+    });
+
+    test('Initial state should be disabled', () => {
+      assert.ok(SyncStateMachine.current, 'disabled');
+    });
+  });
+
+  suite('Events invalid for state', () => {
+    [{
+      current: 'disabled',
+      invalidEvents: [
+        'disable',
+        'sync',
+        'error',
+        'success'
+      ],
+      next: 'enable'
```

Comment:
since 'disabled' is a state, but 'enable' is a transition, maybe 'current' and 'next' are confusing variable names here.